### PR TITLE
to rebase

### PR DESCRIPTION
**Pull request by dangminhtruong**
Xong layouts, them repo

Layout

stash

suggest product

### CODE REVIEW

File: public/js/framgia_script.js [hunk 1]
```diff
@@ -1,7 +1,30 @@
...
     }
 });
+
+$(document).on('click', '.browse', function () {
+    var file = $(this).parent().parent().parent().find('.file');
+    file.trigger('click');
+});
+
+$(document).on('change', '.file', function () {
+    $(this).parent().find('.form-control').val($(this).val().replace(/C:\\fakepath\\/i, ''));
```
Comment (nguyen-thanh-hai) [inline public/js/framgia_script.js]: Xem lại nhé!

File: routes/web.php [hunk 2]
```diff
@@ -14,17 +14,24 @@
...
+
+Route::get('request-fish-tanks-blueprint', 'BlueprintController@getRequestFishTanksBlueprint')
+    ->name('getRequestFishTanksBlueprint');
+Route::post('request-fish-tanks-blueprint', 'BlueprintController@postRequestFishTanksBlueprint')
+    ->name('postRequestFishTanksBlueprint');
+Route::get('create-blueprint', 'BlueprintController@getCreateBlueprint')
+    ->name('getCreateBlueprint');
+Route::post('create-blueprint', 'BlueprintController@postCreateBlueprint')
+    ->name('postCreateBlueprint');
+
```
Comment (nguyen-thanh-hai) [inline routes/web.php]: Xóa trắng + để group lại :D

File: app/Http/Requests/RequireBlueprintRequest.php [hunk 3]
```diff
@@ -24,18 +24,14 @@ public function authorize()
...
         ];
     }
 
     public function messages()
     {
         return [
-            'customer_phone.unique' => 'Your phone number has already taken !',
-            'customer_email.unique' => 'Your email has already taken !',
-            'customer_retype_password' => 'Retype password not matches'
+            'customer_description.required' => 'Vui lòng không để trống'
```
Comment (bsigngz) [inline app/Http/Requests/RequireBlueprintRequest.php]: i18n

File: app/Providers/RepositoryServiceProvider.php [hunk 4]
```diff
@@ -14,17 +17,29 @@ class RepositoryServiceProvider extends ServiceProvider
         'product' => [
             \App\Repositories\Contracts\ProductRepositoryInterface::class,
             \App\Repositories\Eloquents\ProductRepository::class,
+        ],
+        'blueprint' => [
```
Comment (bsigngz) [inline app/Providers/RepositoryServiceProvider.php]: `'blueprint' => [BlueprintRepositoryInterface::class, BlueprintRepository::class],`

File: resources/views/blueprint/create_blueprint.blade.php [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+                <div class="row">
+                    <div class="col-sm-8 col-sm-offset-2 col-md-6 col-md-offset-3">
+                        <h2>{{ __('CreateBlueprint.MainTitle') }}</h2>
+                        <p>{{ __('CreateBlueprint.Title.pTag')  }}}.</p>
+                    </div>
+                </div>
+            </div>
+            <div class="breadcrumb-wrapper">
+                <ol class="breadcrumb">
+                    <li><a href="#">Home</a></li>
```
Comment (bsigngz) [inline resources/views/blueprint/create_blueprint.blade.php]: i18n

File: app/Providers/RepositoryServiceProvider.php (hunk 4 above)
Comment (dangminhtruong) [reply app/Providers/RepositoryServiceProvider.php]: use App\Repositories\Contracts\BlueprintRepositoryInterface;
use App\Repositories\Eloquents\BlueprintRepository;
Em bỏ hai dòng ấy rồi ạ...

**Comment by bsigngz** [comment]
OK

**Pull request merged**